conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -178,19 +187,62 @@ def _get_argument(argument_name, cppinfo_name):
                                        _get_argument("datarootdir", "resdirs")])
         return [el for el in configure_install_flags if el]
 
-    def _default_autoreconf_flags(self):
+    @staticmethod
+    def _default_autoreconf_flags():
         return ["--force", "--install"]
 
+    def _get_triplets(self):
+        triplets = []
+        for flag, value in (("--host=", self._host), ("--build=", self._build),
+                            ("--target=", self._target)):
+            if value:
+                triplets.append(f'{flag}{value}')
+        return triplets
+
+    def update_configure_args(self, **updated_flags):
+        """
+        Helper to update/prune flags from ``self.configure_args``.
+
+        :param updated_flags: ``dict`` with arguments as keys and their argument values.
+                              Notice that if argument value is ``None``, this one will be pruned.
+        """
+        self._update_flags("configure_args", **updated_flags)
+
+    def update_make_args(self, **updated_flags):
+        """
+        Helper to update/prune arguments from ``self.make_args``.
+
+        :param updated_flags: ``dict`` with arguments as keys and their argument values.
+                              Notice that if argument value is ``None``, this one will be pruned.
+        """
+        self._update_flags("make_args", **updated_flags)
+
+    def update_autoreconf_args(self, **updated_flags):
+        """
+        Helper to update/prune arguments from ``self.autoreconf_args``.
+
+        :param updated_flags: ``dict`` with arguments as keys and their argument values.
+                              Notice that if argument value is ``None``, this one will be pruned.
+        """
+        self._update_flags("autoreconf_args", **updated_flags)
+
+    def _update_flags(self, attr_name, **updated_flags):
+        _new_flags = []
+        self_args = getattr(self, attr_name)
+        for index, flag in enumerate(self_args):
+            flag_name = flag.split("=")[0].replace("--", "")
```

Comment:
there is a potential flaw for `--` argument (see https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html)
```
The argument -- terminates all options; any following arguments are treated as non-option arguments, even if they begin with a hyphen.
```
as `"" in something` always evaluates to True